File: Baekjoon_Java/프로그래머스/lv3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean [] visited = new boolean[n];
        for(int i=0; i<n; i++){
            if(visited[i]) continue;
            answer++;
            dfs(i, computers, visited);
        }
        return answer;
    }
    void dfs(int start, int[][]computers, boolean [] visited){
        visited[start] = true;
        for(int i=0; i<computers.length; i++){
            if(visited[i]==false && computers[start][i]==1)
                dfs(i, computers, visited);
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    network = 0
    queue = deque()
    visited = [False] * n 
    
    for i in range(n):
        if not visited[i]:
            queue.append(i)
            visited[i] = True
            
            while queue:
                num = queue.popleft()
                
                for j in range(n):
                    if computers[num][j] == 1 and not visited[j]:
                        queue.append(j)
                        visited[j] = True
            
            network += 1
    
    return network
            
                